harness/determined/estimator/_estimator_context.py

```diff
@@ -60,7 +68,7 @@ def wrap_optimizer(self, optimizer: Any) -> Any:
             return optimizer
 
         self.optimizer_initialized = True
-        if not self.hvd_config.use:
+        if not self.distributed.backend == "horovod":
```

Comment:
does `distributed.backend == "horovod"` indicate:
1. that we are using horovod for dtrain, and we are currently running under horovodrun, or
2. that we are using horovod for dtrain, even though we may be on a single gpu at the moment (no horovodrun)?

Also, what happens if the backend is something other than horovod, but we are in distributed training?  I'm in a rush with this review, but do you assert that EstimatorTrial only works with horovod somewhere?